src/content/users/ProfileView.tsx

```diff
@@ -460,9 +496,14 @@ const ProfileView: FC<Props> = ({ _id, viewType }: Props) => {
                             </StyledTag>
                           );
                         }}
-                        options={studyOptions}
+                        options={
```

Comment:
I do have one potential suggestion to fix your issue when switching from other roles to "Federal Lead". It seems to be a timing issue, Instead of assigning the "All" option at the time of role change, you can instead do it after sorting the study options which happens after every role change anyways due to the addition of the "All" option.

This was just a rough suggestion, if you find any issues with it feel free not to use it.

```javascript
const handleRoleChange = (field: ControllerRenderProps<FormInput, "role">, value: UserRole) => {
    if (prevRoleRef.current === "Federal Lead") {
      setValue(
        "studies",
        studiesField.filter((v) => v !== ALL_STUDIES_OPTION)
      );
    } else if (value === "Federal Lead") {
      setValue("studies", []); // updated line
    }

    field.onChange(value);
  };
  
  ...

useEffect(() => {
    if (fieldset.studies !== "UNLOCKED") {
      return;
    }
    sortStudyOptions();

    if (roleField === "Federal Lead") {
      setValue("studies", [ALL_STUDIES_OPTION]);
    }
  }, [formattedStudyMap, roleField]);
  
  ...
  
  <StyledAutocomplete
     ...
     options={studyOptions}
     ...
```

Reply:
Thanks for looking into this. Your solution seems to work perfectly and it's much cleaner. Updated in 9173800.

Reply:
Applied that change in 7f729d5. I rechecked all of the different scenarios and everything seems to work as expected now.

LMK if you find anything else.